compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/SemanticModel.java

```diff
@@ -58,6 +59,16 @@
      */
     List<Symbol> moduleLevelSymbols();
 
+    /**
+     * If there's an identifier associated with a symbol at the specified cursor position, finds all the references of
+     * the specified symbol within the relevant scope.
+     *
+     * @param fileName name of the file in which to look up the specified position
+     * @param position a cursor position in the source
+     * @return A {@link List} of line ranges of all the references
+     */
+    List<Location> references(String fileName, LinePosition position);
```

Comment:
I feel accepting a symbol as a parameter would make more sense in-terms of the API's perspective..
Then it reads as "give me the references of this symbol"

Reply:
Added an overloaded version of this which takes a `Symbol`.